### DISCUSSION TITLE
What would be the best way to get the map of a city/streets?

### DISCUSSION THREAD

**Discussion by juanchiparra** [d3/d3#3942] [Q&A] [answered]
I have already created maps of countries with their respective states, but I never knew how to do it with cities and their streets. I would understand that you have to start by getting a JSON/TopoJSON, but of the generators that I have gotten none has given me the result I wanted.

I want to recreate such a map, use the data I have gotten with Strava (jogging), and combine them... **What would be the best way to achieve this?** What I want to achieve is to get at least the Strava data to match the map I've generated.

**Thanks!**

![caracas](https://github.com/user-attachments/assets/df8e21f0-7c25-4940-9325-2419d3d6c7db)

**Comment by Fil** [comment, marked as answer]
Hard to say what is the "best" approach, there are quite a few excellent map libraries out there, like Leaflet, Deck.gl, mapbox, maplibre… The street data is often requested as "tiles" so that you only load the data that matches the viewport. There are different flavors of tiles — in terms of formats and access controls. 

Here are a few examples:

* https://observablehq.com/framework/lib/leaflet (using OpenStreetMap tiles)
* https://observablehq.observablehq.cloud/pangea/party/maplibre-gl (OSM tiles, too)
* https://observablehq.observablehq.cloud/pangea/party/maplibre-pmtiles (using protomaps tiles, need self-hosting)
* https://observablehq.com/framework/lib/deckgl (no tiles in this example, but the deckgl website has further examples with tiles)
* https://observablehq.com/framework/lib/mapbox-gl (using mapbox tiles with an API key)

**Reply by juanchiparra** [reply]
Thanks for your response. I'll keep an eye on them
